### ISSUE TITLE
Wireguard-go not allowing connection to intranet

### ISSUE THREAD

**Issue by ErwinP** [boostchicken-dev/udm-utilities#328]
Hi,

Thanks a bunch for providing such [a wonderful tutorial](https://github.com/boostchicken-dev/udm-utilities/wiki/Run-a-Wireguard-VPN-server-on-UDM-Pro) to enable Wireguard on the UDM. I've successfully managed to get it running with my Android phone as a client. I can see the connection using `wg show` and when I check my IP address on my phone, it is the same public address as my UDM (all trafic goes trough the tunnel). However: I cannot access (or even ping) any of my other devices.

I suppose one of the firewall rules is not behaving as expected, but I've checked every step umpf-time, and it seems I configured things as indicated. 

However, one big deviation is the fact that on the UDM, I cannot set a VLAN number, it automatically sets to 2. So I changed everything in the tutorial from 240 to 2:

- All my devices are in `192.168.1.0/24`
- The VPN VLAN is in `192.165.2.0/24`
- The firewall IP address group has the address `192.168.2.1`
- the RFC1918 networks group was left at including `192.168.0.0/16`, but it doesn't help if I specify `192.168.1.0/24` and `192.168.2.0/24` either.

I have no idea how to troubleshoot this, and was wondering if anybody here could offer some help? It would be highly appreciated. Don't hesitate if I should provide more information.

Thank you in advance,
Erwin

**Comment by rohkaemper** [comment]
Just two things:
I guess the VPN VLAN is 192.168. not 192.165. and it is a typo in your description.

Did you set your Allowed IPs correct? Allowed IPs sets routes and permissions to target hosts. So if this is missing it might be an issue and the cause for your inability to connect or ping the hosts.

Hope this helps

**Comment by alxwolf** [comment]
Hallo @ErwinP,

your config file on the second **client** should look like this:

```
[Interface]
PrivateKey = u<the private client key goes here>=
Address = 192.168.2.3/32
DNS = 192.168.2.1

[Peer]
PublicKey = u<the public client key goes here>=
AllowedIPs = 192.168.0.0/16, 10.0.0.0/8, 172.16.0.0/12
Endpoint = <your public server IP / DNS name goes here>:51820
```

basically like the Android file but with a different "AllowedIPs" (192.168.2.3 instead of 192.168.2.2).

On the **server**, this client needs to be added, too: first part is same same no different. 

```
[Interface]
PrivateKey=c<your private server key here>I=
ListenPort = 51820

#Peer Number 1
[Peer]
PublicKey = O<peer number one's public key here>=
AllowedIPs = 192.168.2.2/32
```

then append the block for Peer Number 2:
```
#Peer Number 2
[Peer]
PublicKey = O<peer number one's public key here>=
AllowedIPs = 192.168.2.3/32
```

Reload config and it should hopefully work fine.

```
podman exec -it wireguard wg setconf wg0 /etc/wireguard/wg0.conf

```

**Comment by alxwolf** [comment]
Hallo @ErwinP,

probably something is stuck around your [firewall rule #2](https://github.com/boostchicken-dev/udm-utilities/wiki/Run-a-Wireguard-VPN-server-on-UDM-Pro#rule-2-lan-in---allow-external-traffic-to-vpn-server).

In a first step, you could enable logging for this rule and check via 

```
dmesg
```

if packets are stuck in your firewall.

**Comment by ErwinP** [comment]
Hi,

Thank you both for you help.

At one point, I just removed the last two firewall rules and let the thing rest for a while. Last week I configured them again, and they worked like a charm. I'm not sure why it didn't work at my first attemps - I'm inclined to say that everything was configured correctly as I checked and re-checked an kept on checking, but on the other hand, a mistake from my side is the most reasonable explanation. One thing I did notice, is that the UDM underwent some updates in the meantime. My ego is therefore telling me that something changed due to that :)

Anyhow, I learned a lot by troubleshooting and your comments, so thank you for this valuable input.

Kind regards,
Erwin
